### PR TITLE
Fix #32336 - Checkout Shipping form cache don't get empty

### LINKED ISSUE #32336: Checkout Shipping form cache don't get empty [Magento 2.4.2]
### Preconditions (*)

1. Magento CE 2.4.2 with sample data is installed.
2. Tested on 2.4.0, 2.3.5. Issue propably not reproducible in older versions than 2.4.0.

### Steps to reproduce (*)

1. Add Product to cart
2. Go to checkout
3. Fill shipping form with random data
4. Delete data from Address line 1.
5. delete data from Address line 2.
6. Refresh page.

### Expected result (*)

1. I expect, that after refreshing form, deleted data will not occur on form after refreshing.
2. I expect, that after refreshing page, mage-cache-storage with checkout-data shippingAddressFormData will have empty values.

### Actual result (*)

1. Because of some reason, after filling Address line 1 and deleting it's value, it will break caching of all inputs of this form. 
It is more visible on screenshots below:

2. Filling shipping form: 
Screenshot from 2021-03-02 16-46-52
3. Deleting Data from Address, first line 1 , next line 2:
Screenshot from 2021-03-02 16-49-54
4. Deleting Data from Address first line 2, next line 1:
Screenshot from 2021-03-02 16-50-57
5. Keep updating Address line 1 on keyup till last letter in input:
Screenshot from 2021-03-02 16-53-32

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#32336

### Manual testing scenarios (*)

Please check fixed issue for testing scenario

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run Functional Tests EE

File: app/code/Magento/Checkout/view/frontend/web/js/view/shipping.js [hunk 1]
```diff
@@ -121,9 +121,7 @@ define([
                     );
                 }
                 checkoutProvider.on('shippingAddress', function (shippingAddrsData) {
-                    if (shippingAddrsData.street && !_.isEmpty(shippingAddrsData.street[0])) {
-                        checkoutData.setShippingAddressFromData(shippingAddrsData);
-                    }
+                    checkoutData.setShippingAddressFromData(shippingAddrsData);
```
Comment (Den4ik) [inline app/code/Magento/Checkout/view/frontend/web/js/view/shipping.js:124]: As I see result looks like revert changes for "MC-38048: Incorrect default country displayed on shipping page when store view is changed in cart. Part 2" https://github.com/magento/magento2/commit/d59ad58d924f50350908ae504bbff46ebdef59d6
As result we revert issue with store view change. Could you please recheck it.
Special thanks for mftf.

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @Bartlomiejsz Thanks for your contribution.
Please look at my comments.

**Comment by ishakhsuvarov** [comment]
Hi @Bartlomiejsz 

I am closing this PR for now as it seems like it has become stale and still needs work.

Please feel free to reopen if you wish to continue.

Thanks!

**Pull request closed without merging**